src/overlays/actors/ovl_En_Go/z_en_go.c

```diff
@@ -567,205 +855,236 @@ s32 func_80A122EC(EnGo* this) {
+        if (!(this->actionFlags & ENGO_FLAG_ROLLED_UP)) {
+            this->actionFlags |= ENGO_FLAG_ENGAGED;
         }
     }
 
-    if ((ENGO_GET_F(&this->actor) == ENGO_F_5) || (ENGO_GET_F(&this->actor) == ENGO_F_6) ||
-        (ENGO_GET_F(&this->actor) == ENGO_F_7)) {
-        this->unk_3BC = 0;
-        this->unk_3BE = 0;
-        this->unk_390 |= 0x20;
+    if ((ENGO_GET_TYPE(&this->actor) == ENGO_ASIDE_STORE) || (ENGO_GET_TYPE(&this->actor) == ENGO_ASIDE_ELDERSROOM) ||
+        (ENGO_GET_TYPE(&this->actor) == ENGO_ASIDE_ELDERSSON)) {
+        this->blinkTimer = 0;
+        this->eyeTexIndex = ENGO_EYETEX_OPEN;
+        this->actionFlags |= ENGO_FLAG_EYES_OPEN;
     }
 
-    SubS_UpdateFlags(&this->unk_390, 0, 7);
-    this->unk_3C0 = 0;
-    this->unk_3C4 = 0;
-    this->unk_18C = this->actionFunc;
-    this->actionFunc = func_80A157C4;
+    SubS_UpdateFlags(&this->actionFlags, 0, 7);
+    this->cutsceneState = 0;
+    this->gatekeeperAnimState = 0;
+    this->interruptedActionFunc = this->actionFunc;
+    this->actionFunc = EnGo_Talk;
     return true;
 }
 
-s32 func_80A12868(EnGo* this, PlayState* play) {
-    this->unk_390 &= ~0x800;
-    this->unk_390 &= ~0x1000;
-    this->unk_390 &= ~0x2000;
+/**
+ * Detect any collisions, and convert detections to object specific 'actionFlags'.
+ */
+s32 EnGo_DetectCollisions(EnGo* this, PlayState* play) {
+    this->actionFlags &= ~ENGO_FLAG_HIT_OTHER;
+    this->actionFlags &= ~ENGO_FLAG_HIT_BY_OTHER;
+    this->actionFlags &= ~ENGO_FLAG_HIT_OBJ;
 
     if ((this->colliderCylinder.base.atFlags & AT_HIT) || (this->colliderSphere.base.atFlags & AT_HIT)) {
         this->colliderCylinder.base.atFlags &= ~AT_HIT;
         this->colliderSphere.base.atFlags &= ~AT_HIT;
-        this->unk_390 |= 0x800;
-        this->unk_3C6 = 0x28;
+        this->actionFlags |= ENGO_FLAG_HIT_OTHER;
+        this->harmlessTimer = 40;
     }
 
     if ((this->colliderCylinder.base.acFlags & AC_HIT) || (this->colliderSphere.base.acFlags & AC_HIT)) {
         this->colliderCylinder.base.acFlags &= ~AC_HIT;
         this->colliderSphere.base.acFlags &= ~AC_HIT;
-        this->unk_390 |= 0x1000;
+        this->actionFlags |= ENGO_FLAG_HIT_BY_OTHER;
     }
 
     if ((this->colliderCylinder.base.ocFlags1 & OC1_HIT) || (this->colliderSphere.base.ocFlags1 & OC1_HIT)) {
         this->colliderCylinder.base.ocFlags1 &= ~OC1_HIT;
         this->colliderSphere.base.ocFlags1 &= ~OC1_HIT;
-        this->unk_390 |= 0x2000;
+        this->actionFlags |= ENGO_FLAG_HIT_OBJ;
     }
 
     return false;
 }
 
-s32 func_80A12954(EnGo* this, PlayState* play) {
-    if ((ENGO_GET_F(&this->actor) == ENGO_F_4) && (play->csCtx.state != CS_STATE_IDLE) && (this->actor.draw != NULL) &&
+/**
+ * Start, or finish, the cutscene for the arrival of springtime.
+ */
+s32 EnGo_UpdateSpringArrivalCutscene(EnGo* this, PlayState* play) {
+    if ((ENGO_GET_TYPE(&this->actor) == ENGO_GRAVEYARD) && (play->csCtx.state != 0) && (this->actor.draw != NULL) &&
```

Comment:
```suggestion
    if ((ENGO_GET_TYPE(&this->actor) == ENGO_GRAVEYARD) && (play->csCtx.state != CS_STATE_IDLE) && (this->actor.draw != NULL) &&
```
Seemed to have been dropped during the merge